compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1112,12 +1113,27 @@ public void visit(BLangTypeInit cIExpr) {
                 break;
             case TypeTags.STREAM:
             case TypeTags.CHANNEL:
-                if (cIExpr.initInvocation.argExprs.size() > 0) {
+                if (!cIExpr.initInvocation.argExprs.isEmpty()) {
                     dlog.error(cIExpr.pos, DiagnosticCode.TOO_MANY_ARGS_FUNC_CALL, cIExpr.initInvocation.name);
                     resultType = symTable.semanticError;
                     return;
                 }
                 break;
+            case TypeTags.UNION:
+                List<BType> matchingMembers = findMembersWithMatchingInitFunc(cIExpr, (BUnionType) actualType);
```

Comment:
What if I have union type with objecttype and non object type like map ?

Reply:
If we have `Object1|Object2|map<string> variable = new(args..);`
And if arguments can match one of the `__init` functions of either Object1 or Object2 relevant __init will be called. Otherwise error similar to `cannot infer type of the object from 'Obj|Obj2|Obj3|Obj4'` will be emitted.

We already had type inference support for record/map literals. `Obj|map<string> objAr = { item: "item line" };` this code works without this PR.